src/posix/platform/hdlc_interface.hpp

```diff
@@ -0,0 +1,194 @@
+     * @note This method should be called before reading and sending frames to the interface.
+     *
+     *
+     * @param[in]   aRadioFile    The path to either a UART device or an executable.
+     * @param[in]   aRadioConfig  Parameters to be given to the device or executable.
+     *
+     * @retval OT_ERROR_NONE          The interface is initialized successfully
+     * @retval OT_ERROR_ALREADY       The interface is already initialized.
+     * @retval OT_ERROR_INVALID_ARGS  The UART device or executable cannot be found or failed to open/run.
+     *
+     */
+    otError Init(const char *aRadioFile, const char *aRadioConfig);
+
+    /**
+     * This method deinitializes the interface to the RCP.
+     *
+     */
+    void Deinit(void);
+
+    /**
+     *
+     * This method returns the socket file descriptor associate with the interface
+     *
+     * @returns The associated socket file descriptor, or -1 if interface is not initializes.
+     *
+     */
+    int GetSocket(void) const { return mSockFd; }
+
+    /**
+     * This method indicates whether the `HdclInterface` is currently decoding a received frame or not.
+     *
+     * @returns  TRUE if currently decoding a received frame, FALSE otherwise.
+     *
+     */
+    bool IsDecoding(void) const { return mIsDecoding; }
+
+    /**
+     * This method instructs `HdlcInterface` to read and decode data from radio over the socket.
+     *
+     * If a full HDLC frame is decoded while reading data, this method invokes the `HandleReceivedFrame()` (on the
+     * `aCallback` object from constructor) to pass the received frame to be processed.
+     *
+     */
+    void Read(void);
+
+    /**
+     * This method encodes and sends a frame to Radio Co-processor (RCP) over the socket.
+     *
+     * @param[in] aFrame  A pointer to buffer containing the frame to send.
+     * @param[in] aLength The length (number of bytes) in the frame
+     *
+     * @retval OT_ERROR_NONE     Successfully encoded and sent the frame.
+     * @retval OT_ERROR_NO_BUFS  Insufficient buffer space available to encode the frame.
+     * @retval OT_ERROR_FAILED   Failed to send frame due to socket write failure.
+     *
+     */
+    otError SendFrame(const uint8_t *aFrame, uint16_t aLength);
+
+#if OPENTHREAD_POSIX_VIRTUAL_TIME
+    /**
+     * This method process read data (decode the data).
+     *
+     * This method is intended only for virtual time simulation. Its behavior is similar to `Read()` but instead of
+     * reading the data from the radio socket, it uses the given data in the buffer `aBuffer`.
+     *
+     * @param[in] aBuffer  A pointer to buffer containing data.
+     * @param[in] aLength  The length (number of bytes) in the buffer.
+     *
+     */
+    void ProcessReadData(const uint8_t *aBuffer, uint16_t aLength) { Decode(aBuffer, aLength); }
+#endif
+
+private:
+    enum
+    {
+        kMaxSpinelFrame = 2048,
+    };
+
+    class EncoderBuffer : public Hdlc::Encoder::BufferWriteIterator
```

Comment:
will keep `EncoderBuffer` in `.cpp` be better for compiler performance?

Reply:
I think it won't impact much since it's just a definition (no allocation here) and it's a private one within `HdlcInterface` class.

Reply:
Yes, since it's painful to build wpantund on raspberry PI, I'd like to make header files smaller when possible.

Reply:
Will move it to `.cpp`. It can help a little bit I guess. 
I think boost header files are the main contributor to `wpantund`'s longer build time.